project/cfmc30/README.rst

```diff
@@ -0,0 +1,124 @@
+These transformations decompose signals or images into their frequency components.
+While Fast Fourier Transformation (FFT) has optimized the DFT to an O(nlog(n)) complexity,
+an analogous efficient method for DCT is still a necessity for modern applications.
+
+The DCT is widely used in image compression(JPEG), audio compression(MP3), and other signal
+processing tasks. However, as data sizes grow, performing DCT on high-resolution images or
+large datasets becomes vomputationaly expensive.
+
+FDCT provides a fast DCT algorithm that brings down the time complexity to O(nlog(n)) using
+parallelism and SIMD to take advantage of modern multicore processors.
+
+
+Prospective Users
+=================
+
+FDCP will be valuable in fieds where DCT is used extensively:
+
+* Image Compression
+* Audio Compression
+* Video Processing
+* Signal Processing
+
+System Architecture
+===================
+
+FDCT is built as a Python extension using C++, and interacts with Python codebases
+through Numpy arrays as the primary data structure. The key architectural elements
+are:
+
+#. Input/Output
+
+   * FDCT accepts a 1D or 2D NumPy array as input, where each element represents a
+     pixel intensity (for images) or a signal value.
+
+   * The library outputs the DCT result as another NumPy array of the same dimensions.
+
+#. Configuration Options:
+
+   * Multithreading: Users can enable or disable multithreading,
+     depending on their hardware capabilities.
+
+   * SIMD Optimization: The library allows specifying whether SIMD support is enabled,
+     taking advantage of CPU architectures that provide these instructions (e.g. AVX or SSE).
+
+
+API Description
+===============
+
+Python API
+----------
+
+FDCT will provide a Python interface that integrates with the NumPy library,
+making it easy for Python developers to perform discrete cosine transformations.
+Here is a simple example::
+  import numpy as np
+  import fdct
+
+  # Create a 2D NumPy array (e.g., an image)
+  data = np.random.rand(256, 256)
+
+  # Perform 2D Discrete Cosine Transformation
+  dct_result = fdct.dct2(data, multithread=True, simd=True)
+
+  # Perform Inverse DCT
+  idct_result = fdct.idct2(dct_result)
+
+  # Display the result
+  print(idct_result)
+
+
+Engineering Infrastructure
+==========================
+
+1. Automatic build system
+   This project use CMake to accomplish automatic build system, and also use GitHub
+   CI to automatic build and test the library at each commit.
+2. Version control
+   Git is the main version control tool for this project.
+   After complete main archtecture of the project. Each functionality will be implemented
+   in a branch. After complete of implementation, it will be merged to the dev branch.
+   The stable branch will merge when the code in dev branch is fully tested.
+3. Testing framework
+4. Documentation
+
+
+Schedule
+========
+
+* Planning phase (09/16-10/07)
+* Week 1 (10/14): Basic project infrastructure construction
```

Comment:
Add testing and CI to your development plan.  They should be available early in your development process and help you improve productivity.

Planning for the testing also help you modularize your system design.

Reply:
CI added in planning phase.